### DISCUSSION TITLE
Could homebrew automatically set xattr for Unison?

### DISCUSSION THREAD

**Discussion by reagle** [Homebrew/discussions#3204] [Casks]
Whenever I get a Unison.app update, I have to `xattr -cr /Applications/Unison.app`. I don't know what this means or why it is necessary, but Is it possible for the Homebrew install to do this itself, or at least tell the user to do so if it doesn't already?

**Comment by SMillerDev** [comment]
I think this is covered in https://docs.brew.sh/FAQ#why-cant-i-open-a-mac-app-from-an-unidentified-developer

**Comment by reagle** [comment]
I've encountered three distinct error messages when opening a cask, each with their own intervention. I'm speaking of the third.

1. Are you sure you want to open? Click open and you are good.
2. "“*.app” cannot be opened because the developer cannot be verified." This requires a trip to the security panel in System preferences.
3. "“Unison.app” is damaged and can’t be opened. You should move it to the Trash." This requires `xattr`.

**Reply by SMillerDev** [reply]
That last one generally means the signature is invalid.

**Reply by reagle** [reply]
This has been the case for a long time, for reasons I don't know. I'm suggestion that if the actual problem can't be fixed, perhaps homebrew tell the user that the `xattr` command will be necessary.

**Reply by SMillerDev** [reply]
It might be nicer (also for users outside of homebrew) to ask the developers to check the code signing

**Reply by reagle** [reply]
They have an [open issue](https://github.com/bcpierce00/unison/issues/469) but I don't understand and this has been a problem for a long time. Thus, I think it's best to give homebrew a warning/tip.

**Reply by SMillerDev** [reply]
For some context, the command you're using there basically tells macOS that the code it's about to run is trusted without any reservations. It disables the security attributes that homebrew sets to indicate that the file was downloaded from an unverified source.

While disabling security features might mitigate your issue I'm strongly against homebrew recommending doing this. The only fix for this issue is proper code signing by upstream.

**Reply by reagle** [reply]
It looks like upstream doesn't know how to sign two separate binaries within the same package -- again, I'm just a dumb user trying to be helpful.  Perhaps the brew could point people to the issue at least.
